Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -1103,19 +1207,65 @@ void JitArm64::addex(UGeckoInstruction inst)
   {
     gpr.BindToRegister(d, d == a || d == b);
 
-    // upload the carry state
-    if (!js.carryFlagSet)
+    if (js.carryFlag == CarryFlag::ConstantTrue && !gpr.IsImm(a) && !gpr.IsImm(b))
+    {
+      CMP(ARM64Reg::WZR, ARM64Reg::WZR);
+      js.carryFlag = CarryFlag::InHostCarry;
+    }
+
+    switch (js.carryFlag)
+    {
+    case CarryFlag::InPPCState:
     {
+      // upload the carry state
       ARM64Reg WA = gpr.GetReg();
       LDRB(IndexType::Unsigned, WA, PPC_REG, PPCSTATE_OFF(xer_ca));
       CMP(WA, 1);
       gpr.Unlock(WA);
+      [[fallthrough]];
+    }
+    case CarryFlag::InHostCarry:
+    {
+      if (js.op->wantsCA)
```

Comment:
Isn't this redundant with #9585?

Reply:
Oops, yes. I had a bunch of manual checks like these in this PR and thought I had removed them all after creating PR #9585, but seems like I missed this one.